server/src/test/split.test.ts

```diff
@@ -0,0 +1,65 @@
+const { printReceived, printExpected } = require('jest-matcher-utils');
+import {
+  calculateCompensatorySplit,
+  IDEAL_SPLIT,
+  normalize,
+  randomBucketFromDistribution,
+  Split,
+} from '../lib/model/split';
+
+expect.extend({
+  toAllBeCloseTo(received, argument, precision = 2) {
+    const found = Object.keys(received).find(key => {
+      const pass =
+        Math.abs(received[key] - argument[key]) < Math.pow(10, -precision) / 2;
+      return !pass;
+    });
+
+    return {
+      pass: !found,
+      message: () =>
+        `Expected ${printReceived(received)} to be equal to ${printExpected(
+          argument
+        )}`,
+    };
+  },
+});
+
+describe('calculate compensatory split', () => {
```

Comment:
Love to see these tests! But since you went through all the trouble to create this nice setup, let's test a few more edge cases (all zeros, all 1s, all 0.5s, etc etc). Just want to make sure we're properly handling unexpected input.